test/langtools/jdk/javadoc/doclet/testDiagsLineCaret/MyTaglet.java

```diff
@@ -0,0 +1,145 @@
+
+/**
+ * A taglet that writes messages to the doclet's reporter.
+ */
+public class MyTaglet implements Taglet {
+    private DocletEnvironment docEnv;
+    private Reporter reporter;
+
+    @Override
+    public void init(DocletEnvironment env, Doclet doclet) {
+        Taglet.super.init(env, doclet);
+        docEnv = env;
+
+        // The following should be as simple as
+        //      reporter = ((StandardDoclet) doclet).getReporter();
+        // JDK-8267176
+        try {
+            StandardDoclet sd = (StandardDoclet) doclet;
+            Field htmlDocletField = sd.getClass().getDeclaredField("htmlDoclet");
+            htmlDocletField.setAccessible(true);
+            Object htmlDoclet = htmlDocletField.get(sd);
+            Method getConfigurationMethod = htmlDoclet.getClass().getDeclaredMethod("getConfiguration");
+            Object config = getConfigurationMethod.invoke(htmlDoclet);
+            Method getReporterMethod = config.getClass().getMethod(("getReporter"));
+            reporter = (Reporter) getReporterMethod.invoke(config);
+        } catch (ReflectiveOperationException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    @Override
+    public Set<Location> getAllowedLocations() {
+        return EnumSet.allOf(Location.class);
+    }
+
+    @Override
+    public boolean isInlineTag() {
+        return false;
+    }
+
+    @Override
+    public boolean isBlockTag() {
+        return true;
+    }
+
+    /**
+     * Refines an existing tag ({@code @since} that provides a {@code List<DocTree>},
+     * so that we can better test positions within the tree node. The alternative,
+     * defining a new tag, would use {@code UnknownBlockTagTree} which just provides
+     * a single {@code Doc Tree}.
+     *
+     * @return the name of the tag supported by this taglet
+     */
+    @Override
+    public String getName() {
+        return "since";
+    }
+
+    @Override
+    public String toString(List<? extends DocTree> tags, Element element) {
+        List<Diagnostic.Kind> kinds = Arrays.stream(Diagnostic.Kind.values())
+                .filter(k -> k != Diagnostic.Kind.OTHER)
+                .toList();
+
+        for (Diagnostic.Kind k : kinds) {
+            String message = "This is a " + k.toString().toLowerCase(Locale.ROOT);
+            reporter.print(k, message);
+        }
+
+        for (Diagnostic.Kind k : kinds) {
+            String message = "This is a " + k.toString().toLowerCase(Locale.ROOT) + " for " + element;
+            reporter.print(k, element, message);
+        }
+
+        DocTreePathScanner<Void, Diagnostic.Kind> s = new DocTreePathScanner<>() {
+            @Override
+            public Void scan(DocTree tree, Diagnostic.Kind k) {
+                return super.scan(tree, k);
+            }
+
+            @Override
+            public Void visitEntity(EntityTree node, Diagnostic.Kind k) {
+                if (node.getName().contentEquals("#x1f955")) {
```

Comment:
Is this... a carrot (🥕)?

Reply:
https://www.fileformat.info/info/unicode/char/search.htm :-)